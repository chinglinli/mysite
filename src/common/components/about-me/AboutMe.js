import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'

import styles from './AboutMe.scss'

export default class AboutMe extends React.PureComponent {
  static PropTypes = {
    socialLinks: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        name: PropTypes.string.isRequired,
        url: PropTypes.string.isRequired
      })
    ),
    quote: PropTypes.string
  }

  render () {
    return (
      <div className={styles.root}>
        <div className={styles.profile}>
          <div className={styles.author}/>
          <div className={styles.info}>
            <div>
              <div>我是一隻<em>兩大類</em></div>
              <div>
                {this.props.socialLinks.map(link => (
                  <a key={link.get('name')} className={styles.link} href={link.get('url')}>{link.get('name')}</a>))}
              </div>
            </div>
          </div>
        </div>
        <div className={styles.quote}>{this.props.quote}</div>
      </div>
    )
  }
}
