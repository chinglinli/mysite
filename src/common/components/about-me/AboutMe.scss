@import '../../css/base.scss';

.root {

  .profile {
    height: 90px;

    .author {
      display: inline-block;

      width: 90px;
      height: 90px;
      background: url('./img/marco.jpg') no-repeat center;
      background-size: 90px 90px;

      border: 0.1em solid #FFFFFF;
      box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15);
    }

    .info {
      display: inline-block;
      vertical-align: top;
      margin-top: 20px;
      margin-left: 50px;

      font-size: 1.2em;

      .link {
        @extend .link;
        font-size: 0.7em;

        &:not(:first-child):before {
          content: '|';
          margin: 0 2px;
          color: #6f6f6f;
        }
      }
    }
  }

  .quote {
    margin-top: 25px;
    color: #6f6f6f;
    white-space: pre;

    border-left: 1px solid #ddd;
    padding: .5625em 1.25em 0 1.1875em
  }
}



