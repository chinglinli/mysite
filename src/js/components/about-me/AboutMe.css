.root {

}

.author {
    width: 5em;
    float: left;
    border: 0.1em solid #FFFFFF;
    box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15);
}
