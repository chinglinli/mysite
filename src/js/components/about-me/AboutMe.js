import * as React from 'react';

import styles from './AboutMe.css';
import marcoImg from './img/marco.jpg';

export default class Sidebar extends React.Component {
    render() {
        return (
            <div className={styles.root}>
                <table>
                    <tr>
                        <td><img src={marcoImg} alt='兩大類'/></td>
                        <td>我是一隻<em>兩大類</em>
                        <div id="social">
                            <a href='https://github.com/marco79423'>GitHub</a>&nbsp|&nbsp
                            <a href='https://bitbucket.org/marco79423'>Bitbucket</a>&nbsp|&nbsp
                            <a href='https://www.facebook.com/marco79423'>facebook</a>
                        </div>
                    </td>
                    </tr>
                    <tr>
                        <td colspan='2'>
                            <blockquote>能站著就別坐著，能走路就別騎車<br/>保持站起來的毅力和一步一腳印的耐心。</blockquote>
                        </td>
                    </tr>
                </table>
            </div>
        )
    }
}
