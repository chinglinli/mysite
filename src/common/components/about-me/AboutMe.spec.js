import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import AboutMe from './AboutMe'


test('It should render AboutMe component correctly', () => {
  const config = Immutable.fromJS({
    socialLinks: [
      {name: 'A', url: 'URL 1'},
      {name: 'B', url: 'URL 2'}
    ],
    quote: '今天天氣不錯，希望明天會更好！'
  })

  const component = renderer.create(
    <AboutMe config={config}/>
  )
  expect(component.toJSON()).toMatchSnapshot()
})