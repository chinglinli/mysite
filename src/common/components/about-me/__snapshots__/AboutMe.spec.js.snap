// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render AboutMe component correctly 1`] = `
<div
  className="root"
>
  <div
    className="profile"
  >
    <div
      className="author"
    />
    <div
      className="info"
    >
      <div>
        <div>
          我是一隻
          <em>
            兩大類
          </em>
        </div>
        <div>
          <a
            className="link"
            href="URL 1"
          >
            A
          </a>
          <a
            className="link"
            href="URL 2"
          >
            B
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="quote"
  >
    今天天氣不錯，希望明天會更好！
  </div>
</div>
`;
